### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by george-bitdotio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,358 @@
...
+    }
+
+    // Sleep a bit to not overload the API
+    await sleep(200);
+  }
+});
+```
+
+### Making queries
+
```
Comment (jmo-qap) [inline README.md:86]: Add an opening paragraph enumerating the two options + a 1 sentence tradeoff/indicating which is recommended (possible in the header as well)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,358 @@
...
+#### The `query` method
+
+The `query` runs queries via the bit.io HTTP API, and therefore will have worse
+performance and feature support than direct connections. Importantly, each query run
+via the `query` method is run in a single transaction. The `query` method is recommended
+in situations where installing a database driver is undesirable or impossible, when
+queries are being run very infrequently, or in very short-lived contexts such as one-off
+scripts or serverless backends.
+
+### Connection pooling and management
```
Comment (jmo-qap) [inline README.md:113]:
```suggestion
### Connection pooling and management [important]
```

**Review by jmo-qap — COMMENTED** [review]
Overall LGTM. Few minor comments.

File: README.md (hunk 1 above)
Comment (george-bitdotio) [reply README.md:86]: Done

**Review by jmo-qap — COMMENTED** [review]
Thanks! Over to @eric-bit-io for any details that are node specific

**Review by eric-bit-io — APPROVED** [review]

**Pull request merged**